api/v1/ptpoperatorconfig_types.go

```diff
@@ -58,6 +65,24 @@ type PtpOperatorConfigList struct {
 	Items           []PtpOperatorConfig `json:"items"`
 }
 
+// PtpEventConfig defines the desired state of event framework
+type PtpEventConfig struct {
+	// +kubebuilder:default=false
+	// EnableEventPublisher will deploy event proxy as a sidecar
+	EnableEventPublisher bool `json:"enableEventPublisher,omitempty"`
+	// +kubebuilder:default=5
+	// clock state to stay in holdover state in secs
+	HoldOverTimeout int64 `json:"holdOverTimeout,omitempty"`
+	// +kubebuilder:default=100
+	// max offset in nano secs
+	MaxOffsetThreshold int64 `json:"maxOffsetThreshold,omitempty"`
+	// +kubebuilder:default=-100
+	// min offset in nano secs
+	MinOffsetThreshold int64 `json:"minOffsetThreshold,omitempty"`
+	// transportHost address for event messages.Example amqp://<service-name>.<namespace>.svc.cluster.local
+	TransportHost string `json:"transportHost,omitempty"`
```

Comment:
will this work if the transport is empty? if not we need to remove the omitempty

Reply:
it can be empty if EnableEventPublisher=false. Also if it is empty, the operator will throw an error in the log and it will not deploy the pod.

Reply:
so maybe we don't even need the EnableEventPublisher

lets just make :
EventConfig *PtpEventConfig `json:"ptpEventConfig,omitempty"`

and if it's not nil all the variables most exist

Reply:
fixed: converted to a pointer